### PR TITLE
Add str id's to several twitter entities.

### LINKED ISSUE #794: Add string ID properties for values exceeding IEEE-754 precision limits
### Describe the feature

Since Twitter is already sending us this data, I figured it would be an easy change to implement. 

This is for non 64bit systems/languages or languages that use the [IEEE-754](http://en.wikipedia.org/wiki/IEEE_754-2008) double-precision (looking at you Javascript) that might ingest this scrapers data.

This issue is outlined in twitters documentation here. https://developer.twitter.com/en/docs/twitter-ids

### Would this fix a problem you're experiencing? If so, specify.

This would resolve an issue on my end where we are using SNScrape in a python api but systems ingesting the api are using Nodejs.

### Did you consider other alternatives?

Alternatives would be to apply a transformation on our end (Which we have already done). I figured since Twitter is already sending the data, it probably won't be too much trouble to tack on the extra properties where available.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alexanderolvera**
This is mostly for compatibility with ingestion systems that don't play nice with 64bit integers. 
closes #794 

+ Updated User entity to contain a id_str property.
+ Updated Tweet entity to contain a id_str property.
+ Updated Community entity to contain a id_str property.
+ Small typing fix that was bugging me with the user-agent header

Minor changes from my IDE's formatter. 

Twitters documentation is a little rough when it comes to when the id_str value is and is not available. They claim all 1.1 endpoints contain this property but we are using a v2 endpoint that claims `id` should be a string by default.

https://developer.twitter.com/en/docs/twitter-ids

It seems the v2 endpoint we use is playing by the 1.1 rules so for now I added some fallback to string `id` in case they change it and remove the id_str. (Although looking at how much it is used in this library I feel we would have bigger issues)

### CODE REVIEW

**Comment by TheTechRobo** [comment]
> Minor changes from my IDE's formatter.

snscrape doesn't follow PEP8; I'm not sure if the developer would want these changes or not

File: snscrape/modules/twitter.py [hunk 1]
```diff
@@ -661,7 +663,7 @@ def __init__(self, baseUrl, *, guestTokenManager = None, maxEmptyPages = 0, **kw
 		self._guestTokenManager = guestTokenManager
 		self._maxEmptyPages = maxEmptyPages
 		self._apiHeaders = {
-			'User-Agent': None,
+			'User-Agent': typing.Union[str, None],
```
Comment (TheTechRobo) [inline snscrape/modules/twitter.py:666]: This seems a bit odd, assigning a type to a dictionary

File: snscrape/modules/twitter.py [hunk 2]
```diff
@@ -1719,6 +1748,7 @@ def _get_entity(self):
 		return User(
 			username = user['legacy']['screen_name'],
 			id = int(user['rest_id']),
+			id_str=user['rest_id'],
```
Comment (TheTechRobo) [inline snscrape/modules/twitter.py:1751]: Spaces before and after the equals sign, I believe

File: snscrape/modules/twitter.py [hunk 3]
```diff
@@ -1805,7 +1835,6 @@ def get_items(self):
 			'verified_phone_label_enabled': False,
 			'responsive_web_graphql_timeline_navigation_enabled': True,
 			'responsive_web_graphql_skip_user_profile_image_extensions_enabled': False,
-			'responsive_web_graphql_skip_user_profile_image_extensions_enabled': False,
```
Comment (TheTechRobo) [inline snscrape/modules/twitter.py:1808]: what does removing this do?

**Review by TheTechRobo — CHANGES_REQUESTED** [review]

**Comment by alexanderolvera** [comment]
> > Minor changes from my IDE's formatter.
> 
> snscrape doesn't follow PEP8; I'm not sure if the developer would want these changes or not

I figured as much, Tried to remove as many as possible. I think most of those edits are only on the feature changes.

**Comment by TheTechRobo** [comment]
If they are, it's hard to tell, as there are e.g. indentation changes that make it hard to see the actual changes.

File: snscrape/modules/twitter.py (hunk 3 above)
Comment (alexanderolvera) [reply snscrape/modules/twitter.py:1808]: It was a duplicate of the line above it. Figured I would yeet it while I was in there.

Comment (TheTechRobo) [reply snscrape/modules/twitter.py:1808]: Oh, yes, I didn't see that!

**Comment by JustAnotherArchivist** [comment]
> I'm not sure if the developer would want those changes or not

I do indeed not want any of those stylistic changes. Please revert all of them, then we can discuss the actual functional changes.

**Comment by alexanderolvera** [comment]
> > I'm not sure if the developer would want those changes or not
> 
> I do indeed not want any of those stylistic changes. Please revert all of them, then we can discuss the actual functional changes.

Sounds good. I'll close out this PR to re-implement by changes with the formatter turned off.

**Pull request closed without merging**